File: src/main/java/mk/finki/ukim/wpaud/model/Product.java
package mk.finki.ukim.wpaud.model;

import lombok.Data;
import org.hibernate.annotations.Cascade;

import javax.persistence.*;

@Data
@Entity

public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private Double price;

    private Integer quantity;
    @ManyToOne
    private Category category;
    @ManyToOne
    private Manufacturer manufacturer;

    public Product(String name, Double price, Integer quantity, Category category, Manufacturer manufacturer) {
        //bad practice; will have to be changed when we have a real db
        //this.id = (long) (Math.random() * 1000);
        this.name = name;
        this.price = price;
        this.quantity = quantity;
        this.category = category;
        this.manufacturer = manufacturer;
    }

    public Product() {}
}

File: frontend/src/components/Products/ProductTerm/Product.js
import React from "react";


const product = (props) => {
    return (
        <tr>
            <td scope={"col"}>{props.term.name}</td>
            <td scope={"col"}>{props.term.price}</td>
            <td scope={"col"}>{props.term.quantity}</td>
            <td scope={"col"}>{props.term.category.name}</td>
            {/*treba da e manufacturer ama dava greska*/}
            <td scope={"col"}>{props.term.manufacturer.name}</td>
        </tr>
    );
}

export default product;